source/type/type_utils.h

```diff
@@ -79,6 +79,15 @@ static bt::ptime get_base_dt(const nt::StopTime* st_orig, const nt::StopTime* st
     return bt::ptime(validity_pattern_dt_day, boost::posix_time::seconds(hour_of_day_base));
 }
 
+static const nt::StopTime& earliest_stop_time(const std::vector<nt::StopTime>& sts) {
```

Comment:
as this file is included in type.cpp (which affects pretty much the whole project), it would be nice to put the 4 impl of these functions in a cpp (and remove the static).
avoids unused function warning, and avoids recompiling all when modifying one method